src/World.cc

```diff
@@ -848,9 +877,25 @@ Errors World::Implementation::LoadSphericalCoordinates(
 
   // Create coordinates
   this->sphericalCoordinates.emplace();
-  this->sphericalCoordinates =
-      gz::math::SphericalCoordinates(surfaceModel, latitude, longitude,
-      elevation, heading);
+  if (surfaceModel != gz::math::SphericalCoordinates::CUSTOM_SURFACE)
+  {
+    this->sphericalCoordinates =
+        gz::math::SphericalCoordinates(surfaceModel, latitude, longitude,
+        elevation, heading);
+  }
+  else
+  {
+    auto sphericalCoordinatesObject =
+      gz::math::SphericalCoordinates(surfaceModel,
+          axisEquatorial, axisPolar);
```

Comment:
Can we just use `this->sphericalCoordinates` directly instead of assigning it later on? eg.,
```suggestion
   this->sphericalCoordinates =
      gz::math::SphericalCoordinates(surfaceModel,
          axisEquatorial, axisPolar);
```

(Then calling the set functions on `this->sphericalCoordinates`)

Reply:
Done https://github.com/gazebosim/sdformat/pull/1050/commits/0d9a9556f1554e479e1da13ddace7b6575a6c5e5